rviz_rendering/include/rviz_rendering/objects/screw_visual.hpp

```diff
@@ -0,0 +1,106 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef RVIZ_RENDERING__OBJECTS__SCREW_VISUAL_HPP_
+#define RVIZ_RENDERING__OBJECTS__SCREW_VISUAL_HPP_
+
+#include "rviz_rendering/objects/screw_visual.hpp"
+
+#include <OgreSceneNode.h>
+#include <OgreSceneManager.h>
+
+#include "rviz_rendering/objects/arrow.hpp"
+#include "rviz_rendering/objects/billboard_line.hpp"
+
+#include "rviz_rendering/visibility_control.hpp"
+
+namespace rviz_rendering
+{
+// ScrewVisual visualizes a single screw, i.e. a wrench, twist, or acceleration
+class ScrewVisual
+{
+public:
+  // Constructor.
+  // Creates the visual stuff and puts it into the scene, but in an unconfigured state.
+  RVIZ_RENDERING_PUBLIC
+  ScrewVisual(Ogre::SceneManager * scene_manager, Ogre::SceneNode * parent_node);
+
+  // Destructor.  Removes the visual stuff from the scene.
+  virtual ~ScrewVisual();
+
+  // Configure the visual to show the given linear and angular vectors
+  RVIZ_RENDERING_PUBLIC
+  void setScrew(const Ogre::Vector3 & linear, const Ogre::Vector3 & angular);
+
+  // Set the pose of the coordinate frame the message refers to.
+  // This could be done inside setMessage(), but that would require calls to FrameManager
+  // and error handling inside setMessage(), which doesn't seem as clean.
+  // This way ScrewVisual is only responsible for visualization.
+  RVIZ_RENDERING_PUBLIC
+  void setFramePosition(const Ogre::Vector3 & position);
+  RVIZ_RENDERING_PUBLIC
+  void setFrameOrientation(const Ogre::Quaternion & orientation);
+
+  // Set the color and alpha of the visual, which are user-editable
+  // parameters and therefore don't come from the message.
+  RVIZ_RENDERING_PUBLIC
+  void setLinearColor(float r, float g, float b, float a);
+  RVIZ_RENDERING_PUBLIC
+  void setAngularColor(float r, float g, float b, float a);
+  RVIZ_RENDERING_PUBLIC
+  void setLinearScale(float s);
+  RVIZ_RENDERING_PUBLIC
+  void setAngularScale(float s);
+  RVIZ_RENDERING_PUBLIC
+  void setWidth(float w);
+  RVIZ_RENDERING_PUBLIC
+  void setHideSmallValues(bool h);
+  RVIZ_RENDERING_PUBLIC
+  void setVisible(bool visible);
+
+private:
+  // The object implementing the circle
+  rviz_rendering::Arrow * arrow_linear_;
+  rviz_rendering::Arrow * arrow_angular_;
+  rviz_rendering::BillboardLine * circle_angular_;
+  rviz_rendering::Arrow * circle_arrow_angular_;
+  float linear_scale_, angular_scale_, width_;
```

Comment:
This is definitely a matter of taste, but I find it easier to read if these are on separate lines.  If you disagree, feel free to ignore this:

```suggestion
  float linear_scale_;
  float angular_scale_;
  float width_;
```